master/internal/configpolicy/task_config_policy.go

```diff
@@ -28,7 +31,74 @@ type NTSCConfigPolicies struct {
 	Constraints     *model.Constraints   `json:"constraints"`
 }
 
-// PriorityAllowed returns true if the desired priority is within the limit set by task config policies.
+var (
+	errPriorityConstraintFailure = errors.New("submitted workload failed priority constraint")
+	errResourceConstraintFailure = errors.New("submitted workload failed a resource constraint")
+)
+
+// CheckNTSCConstraints returns true if the NTSC config passes constraint checks.
+func CheckNTSCConstraints(
```

Comment:
Small nit but after reading the function's comment, maybe it would be a little clearer to call this function `ValidNTSCConfig`? esp since it returns a bool!

Reply:
I do like the func name as is though, so absolutely up to u but just a suggestion! 😆